RecoVertex/KinematicFit/interface/KinematicConstrainedVertexUpdatorT.h

```diff
@@ -139,6 +140,14 @@ RefCountedKinematicVertex KinematicConstrainedVertexUpdatorT<nTrk, nConstraint>:
   val += g * delta_alpha;
   lambda = v_g_sym * val;
 
+  for (auto lambda_element : lambda) {
+    if (!edm::isFinite(lambda_element)) {
```

Comment:
according to https://github.com/cms-sw/cmssw/issues/39267#issue-1357282107
the `val` already has `inf` values.
So, in this context, the check can be made earlier.

If the point is that more arithmetic can lead to inf/nan, then shouldn't there be a check at the end

Reply:
doesn't https://github.com/cms-sw/cmssw/issues/39267#issue-1357282107 say that lambda can nan even when val is something finite? 
In that case, it would not be sufficient to only check val. 
I understood that this bugfix will prevent the crashes in the IB, which I understand is urgent.
If it's not really that urgent than we can wait till I (or someone else!) find the time to find the optimal solution, identifying the nan at the earliest point in the code.  
I can't guarantee that this will happen quickly (certainly not this weekend).

Reply:
both examples in #39267 had inf in the `val`

Reply:
> I understood that this bugfix will prevent the crashes in the IB, which I understand is urgent.

well, the earlier fix to #36788 took 7 months and was more extensive.